pinot-broker/src/main/java/org/apache/pinot/broker/routing/v2/segmentselector/SegmentSelector.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.routing.v2.segmentselector;
+
+import java.util.List;
+import java.util.Set;
+import org.apache.helix.model.ExternalView;
+import org.apache.pinot.common.request.BrokerRequest;
+
+
+/**
+ * The segment selector selects the segments for the query. The segments selected should cover the whole dataset (table)
+ * without overlap.
+ */
+public interface SegmentSelector {
```

Comment:
Can we not enhance the existing SegmentSelector?

Reply:
The current `SegmentSelector` is not wired up, and serves different purpose. In the future, that should be integrated into this segment selector.